### DISCUSSION TITLE
Where can I put my third-party script if not in next/head ?

### DISCUSSION THREAD

**Discussion by rscharfer** [vercel/next.js#17358] [Help] [answered]
I am trying to add a [really cool comments widget ](https://utteranc.es/) to my Next.js blog which requires you to place a `script` tag at the place in the DOM where you want your comments widget to be. The script uses `insertAdjecentHTML` to place an `iframe` immediately after itself in the DOM, and then it deletes itself from the DOM. 

If I try to add this script using `next/head`, the iframe will be placed in the `<head>` which obviously won't work. 

How else can I install this?  

When I include the `script` tag as part of my returned JSX,  the script from the `src` is retrieved and executed as it should be, and the new iframe is added to the DOM as it should be, but somehow Next deletes milliseconds afterwards.   There seems to be some sort of race condition going on that I do not completely understand. 

Any ideas?

**Comment by rscharfer** [comment, marked as answer]
The solution to my problem was to manually create a script tag after the component mounted and insert it into a container.  Also, more a React issue, not Next.  See this for details. https://github.com/utterance/utterances/issues/161

**Reply by uinstinct** [reply]
Is adding the next/head better in the index.js or the _app.js (considering the application to be only a single page) ?
